--- a/options.c
+++ b/options.c
@@ -76,6 +76,10 @@
 static int getline_error;
 
 
+#ifndef _PATH_DEFTAPE
+#define _PATH_DEFTAPE "/dev/st0"
+#endif
+
 #define GZIP_CMD	"gzip"		/* command to run as gzip */
 #define COMPRESS_CMD	"compress"	/* command to run as compress */
 #define BZIP2_CMD	"bzip2"		/* command to run as bzip2 */
--- a/tar.1
+++ b/tar.1
@@ -144,7 +144,7 @@
 .It Fl f Ar archive
 Filename where the archive is stored.
 Defaults to
-.Pa /dev/rst0 .
+.Pa /dev/st0 .
 .It Fl H
 Follow symlinks given on the command line only.
 .It Fl h
@@ -279,18 +279,18 @@
 The options
 .Op Fl 014578
 can be used to select one of the compiled-in backup devices,
-.Pa /dev/rstN .
+.Pa /dev/stN .
 .Sh ENVIRONMENT
 .Bl -tag -width Fl
 .It Ev TMPDIR
 Path in which to store temporary files.
 .It Ev TAPE
 Default tape device to use instead of
-.Pa /dev/rst0 .
+.Pa /dev/st0 .
 .El
 .Sh FILES
-.Bl -tag -width "/dev/rst0"
-.It Pa /dev/rst0
+.Bl -tag -width "/dev/st0"
+.It Pa /dev/st0
 default archive name
 .El
 .Sh EXIT STATUS
--- a/tar.h
+++ b/tar.h
@@ -122,12 +122,12 @@
 /*
  * default device names
  */
-#define	DEV_0		"/dev/rst0"
-#define	DEV_1		"/dev/rst1"
-#define	DEV_4		"/dev/rst4"
-#define	DEV_5		"/dev/rst5"
-#define	DEV_7		"/dev/rst7"
-#define	DEV_8		"/dev/rst8"
+#define	DEV_0		"/dev/st0"
+#define	DEV_1		"/dev/st1"
+#define	DEV_4		"/dev/st4"
+#define	DEV_5		"/dev/st5"
+#define	DEV_7		"/dev/st7"
+#define	DEV_8		"/dev/st8"
 #endif /* _PAX_ */
 
 /*
