--- a/cache.c
+++ b/cache.c
@@ -191,7 +191,7 @@
 	 * No entry for this uid, we will add it
 	 */
 	if (!pwopn) {
-		setpassent(1);
+		setpwent();
 		++pwopn;
 	}
 	if (ptr == NULL)
@@ -257,7 +257,7 @@
 	 * No entry for this gid, we will add it
 	 */
 	if (!gropn) {
-		setgroupent(1);
+		setgrent();
 		++gropn;
 	}
 	if (ptr == NULL)
@@ -324,7 +324,7 @@
 	}
 
 	if (!pwopn) {
-		setpassent(1);
+		setpwent();
 		++pwopn;
 	}
 
@@ -387,7 +387,7 @@
 	}
 
 	if (!gropn) {
-		setgroupent(1);
+		setgrent();
 		++gropn;
 	}
 	if (ptr == NULL)
