From a81da58a67f262a0bb09a1b9b2c4f0af2c94d578 Mon Sep 17 00:00:00 2001
From: Eivind Uggedal <eivind@uggedal.com>
Date: Fri, 5 Jun 2015 11:11:09 +0000
Subject: [PATCH] ln: allow dangling symlinks

---
 ln.c | 25 ++++++++++++++-----------
 1 file changed, 14 insertions(+), 11 deletions(-)

diff --git a/ln.c b/ln.c
index 6e40c8f22710..ef3997cec2df 100644
--- a/ln.c
+++ b/ln.c
@@ -65,20 +65,23 @@ main(int argc, char *argv[])
 		if (!hastarget)
 			target = basename(*argv);
 
-		if (stat(*argv, &st) < 0) {
-			weprintf("stat %s:", *argv);
-			ret = 1;
-			continue;
-		} else if (fstatat(dirfd, target, &tst, AT_SYMLINK_NOFOLLOW) < 0) {
-			if (errno != ENOENT) {
-				weprintf("fstatat %s %s:", targetdir, target);
+		if (!sflag) {
+			if (stat(*argv, &st) < 0) {
+				weprintf("stat %s:", *argv);
+				ret = 1;
+				continue;
+			} else if (fstatat(dirfd, target, &tst, AT_SYMLINK_NOFOLLOW) < 0) {
+				if (errno != ENOENT) {
+					weprintf("fstatat %s %s:", targetdir, target);
+					ret = 1;
+					continue;
+				}
+			} else if (st.st_dev == tst.st_dev && st.st_ino == tst.st_ino) {
+				weprintf("%s and %s/%s are the same file\n",
+						*argv, targetdir, target);
 				ret = 1;
 				continue;
 			}
-		} else if (st.st_dev == tst.st_dev && st.st_ino == tst.st_ino) {
-			weprintf("%s and %s/%s are the same file\n", *argv, targetdir, target);
-			ret = 1;
-			continue;
 		}
 
 		if (fflag && unlinkat(dirfd, target, 0) < 0 && errno != ENOENT) {
-- 
2.4.2

