From 30e30cd3081f7a03263d6265f1c9f17fe0804ff0 Mon Sep 17 00:00:00 2001
From: Eivind Uggedal <eivind@uggedal.com>
Date: Tue, 9 Jun 2015 11:42:19 +0000
Subject: [PATCH] printf: flag support for %s

---
 printf.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/printf.c b/printf.c
index 13359067f2a1..e45ca74ab7c5 100644
--- a/printf.c
+++ b/printf.c
@@ -124,7 +124,10 @@ main(int argc, char *argv[])
 			free(rarg);
 			break;
 		case 's':
-			printf("%*.*s", width, precision, arg);
+			fmt = estrdup("%#*.*s");
+			fmt[1] = flag;
+			printf(fmt, width, precision, arg);
+			free(fmt);
 			break;
 		case 'd': case 'i': case 'o': case 'u': case 'x': case 'X':
 			arglen = strlen(arg);
-- 
2.4.2

