From 94df11a7ffefbae7277bc9dea5283aeea2aed919 Mon Sep 17 00:00:00 2001
From: Evan Gates <evan.gates@gmail.com>
Date: Wed, 3 Jun 2015 07:31:31 -0700
Subject: [PATCH] delete pattern space before updating ranges on c,d,D

---
 sed.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/sed.c b/sed.c
index b8e4be2..a5ca4b4 100644
--- a/sed.c
+++ b/sed.c
@@ -1665,6 +1665,7 @@ app_line(void)
 static void
 new_next(void)
 {
+	*patt.str = '\0';
 	update_ranges(pc + 1, prog + pcap);
 	new_line();
 	pc = prog - 1;
-- 
2.4.2

