From 7c382e85fa5aa68aa8b77b3cea268be9c2dd03ec Mon Sep 17 00:00:00 2001
From: Eivind Uggedal <eivind@uggedal.com>
Date: Tue, 19 May 2015 12:20:04 +0000
Subject: [PATCH] uniq: fix for segfault given no args

---
 uniq.c | 29 +++++++++++++----------------
 1 file changed, 13 insertions(+), 16 deletions(-)

diff --git a/uniq.c b/uniq.c
index 922ba3973797..61d47f299567 100644
--- a/uniq.c
+++ b/uniq.c
@@ -121,23 +121,20 @@ main(int argc, char *argv[])
 	if (argc > 2)
 		usage();
 
-	if (!argc) {
-		uniq(stdin, stdout);
-	} else {
-		if (argv[0][0] == '-' && !argv[0][1]) {
-			argv[0] = "<stdin>";
-			fp = stdin;
-		} else if (!(fp = fopen(argv[0], "r"))) {
-			eprintf("fopen %s:", argv[0]);
-		}
-		if (argc == 1 || (argv[1][0] == '-' && !argv[1][1])) {
-			argv[1] = "<stdout>";
-			ofp = stdout;
-		} else if (!(ofp = fopen(argv[1], "w"))) {
-			eprintf("fopen %s:", argv[1]);
-		}
-		uniq(fp, ofp);
+	if (!argc || (argv[0][0] == '-' && !argv[0][1])) {
+		argv[0] = "<stdin>";
+		fp = stdin;
+	} else if (!(fp = fopen(argv[0], "r"))) {
+		eprintf("fopen %s:", argv[0]);
 	}
+	if (!argc || argc == 1 || (argv[1][0] == '-' && !argv[1][1])) {
+		argv[1] = "<stdout>";
+		ofp = stdout;
+	} else if (!(ofp = fopen(argv[1], "w"))) {
+		eprintf("fopen %s:", argv[1]);
+	}
+
+	uniq(fp, ofp);
 	uniqfinish(ofp);
 
 	efshut(fp, argv[0]);
-- 
2.4.0


