--- a/emacs.c
+++ b/emacs.c
@@ -16,7 +16,7 @@
 
 #include "sh.h"
 #include <sys/stat.h>
-#include <sys/queue.h>
+#include <bsd/sys/queue.h>
 #include <ctype.h>
 #include <locale.h>
 #include "edit.h"
--- a/sh.h
+++ b/sh.h
@@ -15,7 +15,7 @@
 #include <setjmp.h>
 #include <stdbool.h>
 #include <stddef.h>
-#include <stdlib.h>
+#include <bsd/stdlib.h>
 #include <unistd.h>
 #include <string.h>
 #include <stdarg.h>
