From 200b75a3f30a98aa531d77a76d804616a2fae53c Mon Sep 17 00:00:00 2001
From: Eivind Uggedal <eivind@uggedal.com>
Date: Tue, 15 Mar 2016 16:02:33 +0000
Subject: [PATCH 1/2] dd: always create of= even if seek= is used

---
 dd.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/dd.c b/dd.c
index 977f0f0a666f..e37c6c7472b2 100644
--- a/dd.c
+++ b/dd.c
@@ -42,7 +42,7 @@ static int
 prepare_copy(struct dd_config *ddc, int *ifd, int *ofd)
 {
 	struct stat st;
-	int fli = O_RDONLY|O_LARGEFILE|O_NOCTTY, flo = O_WRONLY|O_LARGEFILE|O_NOATIME|O_NOCTTY;
+	int fli = O_RDONLY|O_LARGEFILE|O_NOCTTY, flo = O_WRONLY|O_LARGEFILE|O_CREAT|O_NOATIME|O_NOCTTY;
 	uid_t euid = 0;
 	long pagesize;
 
@@ -106,7 +106,7 @@ prepare_copy(struct dd_config *ddc, int *ifd, int *ofd)
 	}
 
 	if (!ddc->seek)
-		flo |= O_CREAT|O_TRUNC;
+		flo |= O_TRUNC;
 
 	if (!ddc->out) *ofd = 1;
 	else if ((*ofd = open(ddc->out, flo, st.st_mode)) < 0) {
-- 
2.1.4

