From 3f9138636e3565c3571d41afe490cfb8b0a31e8c Mon Sep 17 00:00:00 2001
From: Eivind Uggedal <eivind@uggedal.com>
Date: Tue, 15 Mar 2016 20:27:49 +0000
Subject: [PATCH 4/6] dd: don't let failure to change pipe size affect exit code

---
 dd.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/dd.c b/dd.c
index 1e1fda1c28c2..daa75fddecdf 100644
--- a/dd.c
+++ b/dd.c
@@ -152,6 +152,7 @@ copy_splice(struct dd_config *ddc)
 		if (fcntl(p[0], F_SETPIPE_SZ, 1<<n) != -1)
 			break;
 	}
+	errno = 0;
 #endif
 	n = ddc->bs;
 	for (;ddc->b_out != ddc->count && !sigint;) {
-- 
2.1.4

